### DISCUSSION TITLE
Retrieving stored image from Pocketbase

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#4627] [Q&A] [answered]
Hi, Would appreciate any pointers on this.
I have a collection, called 'Student':
![image](https://github.com/pocketbase/pocketbase/assets/12494537/1834d532-3b45-4499-b656-203437fbdc80)

I'm trying to retrieve a single record from this collection, which should contain the actual data of the picture file that's stored in the collection. This record I wish to send in response to a fetch operation by the browser, at which I wish to insert into an `<img>`
 element. However, querying the record merely gives the picture's filename. The code and output is as below:

Code:
`const rawRecord = await locals.userPb.collection('Student').getOne(params.id);
console.log(JSON.stringify(rawRecord,null,2));`

Output:
`{
  "active": true,
  "collectionId": "7vfpthon2jsteho",
  "collectionName": "Student",
  "created": "2024-03-21 08:35:29.861Z",
  "email": "dee@mail.com",
  "gender": "F",
  "id": "3u5oebp4q74n5lx",
  "mobile": "45678",
  "name": "Dee",
  "profilePicture": "13NhPQP4v6V0_MRJKLGcufU.jpg",
  "updated": "2024-03-25 02:30:45.694Z",
  "expand": {}
}
{ pathname: '/students/3u5oebp4q74n5lx/personal' }`

I know there is the scheme of using `pb.files.getUrl()` and using the output of this as the `src` attribute value in the `<img>` element. But I would like the data sent directly in the fetch response. Is there any way to achieve this? 

Many thanks in advance.

**Comment by rameshkumar70** [comment]
Hi, I'm trying another approach. I am using `getUrl()` and fetching (using `fetch()`) the image file located at the URL provided by `getUrl()`. I am now trying to convert the fetched file into a _Base64_ string, using the code below:

`let response = await fetch(picUrl);`
	`let contentType = response.headers.get("Content-Type");`
	`let buffer = await response.arrayBuffer();`
	`const profilePictureBase64 = "data:" + contentType + ';base64,' + buffer.toString('base64');`
	`console.log(profilePictureBase64);`

But instead of a string with the image file data encoded as Base64, I get this output from the last `console.log()` in the above code:

**data:image/png;base64,[object ArrayBuffer]**

I am very close to getting a solution, and I am almost certain this has nothing to do with pocketbase at this point. Can anyone please help shed some light on what I could do. Already at my wits end. Thank you so much.

**Reply by shubhendumadhukar** [reply]
If your goal is to just display the image, you can simply use the image url, i.e. `http://<<YourPbHost>>/api/files/Student/{params.id}/{rawRecord.profilePicture}`

**Reply by rameshkumar70** [reply]
Thank you for the tip. I've got this already working by embedding that URL in the `<img>` tag . But I wish to avoid exposing my pocketbase server and instead have only the svelte-backend accessible directly from the browser.

**Comment by rameshkumar70** [comment, marked as answer]
Finally got something that gave me what I wanted. This is for folks who might come looking for a solution of a problem like what I faced.

1) To summarize, I am running sveltekit (backend) and pocketbase. I wish to have only svelte serving browsers while pocketbase is run internallly, accessible only from the svelte server files.

2) I have an image file stored in each record in a pocketbase collection (Student). I want to display this image using an `<img>` element in the browser. Using pocketbase's `getUrl()` and embedding its output as the `src` attribute value of `<img>`, will require me to have pocketbase server exposed, which is not what I want. 

3) I then used `fetch()` from sveltekit along with the url provided by pocketbase's `getUrl()` instead. The fetch's response is then converted into a _dataURL_ which is then set as the `src` attribute value of the `<img>` element. Below is the code I used, run in the +page.server.js file of sveltekit:



`const rawRecord = await locals.userPb.collection('Student').getOne(params.id);`
`const record = structuredClone(rawRecord);`
`const token = await locals.userPb.files.getToken();`
`const picUrl = locals.userPb.files.getUrl(record, record.profilePicture, { token: token });`
`const response = await fetch(picUrl);`
`const contentType = response.headers.get('Content-Type');`
`const buffer = await response.arrayBuffer();`
`// Send picDataUrl to html page and embed the value in the src attribute of <img> element`
`const picDataUrl = 'data:' + contentType + ';base64,' + Buffer.from(buffer).toString('base64');`

Hope this helps someone.

**Reply by hpsopvopskopefop** [reply]
Thank you so much for documenting your own solution!

**Reply by YasinDaking** [reply]
Nice
